src/acceptor.rs

```diff
@@ -0,0 +1,139 @@
+use core::task::{Context, Poll};
+use std::future::Future;
+use std::io;
+use std::pin::Pin;
+use std::sync::Arc;
+
+use futures_util::ready;
+use hyper::server::{
+    accept::Accept,
+    conn::{AddrIncoming, AddrStream},
+};
+use rustls::ServerConfig;
+use tokio::io::{AsyncRead, AsyncWrite, ReadBuf};
+
+mod builder;
+pub use builder::AcceptorBuilder;
+use builder::WantsTlsConfig;
+
+enum State {
+    Handshaking(tokio_rustls::Accept<AddrStream>),
+    Streaming(tokio_rustls::server::TlsStream<AddrStream>),
+}
+
+// tokio_rustls::server::TlsStream doesn't expose constructor methods,
+// so we have to TlsAcceptor::accept and handshake to have access to it
+// TlsStream implements AsyncRead/AsyncWrite handshaking tokio_rustls::Accept first
```

Comment:
I'm having trouble understanding the second line of this comment, on L26. Does this suggested edit track what you're trying to say? 

```suggestion
// TlsStream implements AsyncRead/AsyncWrite by handshaking with tokio_rustls::Accept first
```